### PR TITLE
Fix condition race in webWorker

### LINKED ISSUE #196: Failure when reporting bundle evaluation error
### Current Behavior
screen shot 2017-07-27 at 10 57 17

### Expected Behavior
No error

### Haul Configuration (webpack.haul.js)
Default

I believe this is due to https://github.com/callstack-io/haul/commit/edff2ce7c8bef4185cdaec8b9668c8a2bb4a8007
CC: @Krizzu

### PR DESCRIPTION
**Pull request by Krizzu**
Fixes #196

### CODE REVIEW

File: src/server/assets/debuggerWorker.js [hunk 1]
```diff
@@ -30,19 +34,34 @@ onmessage = (function() {
...
 
+      shouldQueueMessages = true;
+
       function evalJS(js) {
         try {
-          eval(js);
+          eval(
+            js
+              .replace('this[“webpackHotUpdate”]', 'self[“webpackHotUpdate”]')
+              .replace('this[“webpackHotUpdate”]', 'self[“webpackHotUpdate”]'),
```
Comment (thymikee) [inline src/server/assets/debuggerWorker.js]: why is this doubled?

Comment (Krizzu) [inline src/server/assets/debuggerWorker.js]: Changed that to regexp

File: src/server/assets/debuggerWorker.js [hunk 2]
```diff
@@ -30,19 +34,33 @@ onmessage = (function() {
...
       }
 
+      shouldQueueMessages = true;
+
       function evalJS(js) {
         try {
-          eval(js);
+          eval(
+            js
+              .replace(/this\[“webpackHotUpdate”\]/g, 'self[“webpackHotUpdate”]')
```
Comment (thymikee) [inline src/server/assets/debuggerWorker.js]: Maybe I don't have enough context, but this quote `“` is different than default double quote:`"`. Is this intended?

Comment (Krizzu) [inline src/server/assets/debuggerWorker.js]: Yup it is different, but that does not affect anything. Going to change it anyway :)

**Comment by Krizzu** [comment]
CC @grabbou

**Pull request merged**